### PR TITLE
Hw7 task2

### PR DESCRIPTION
**Pull request by E1-Bart0**

### CODE REVIEW

File: test/hw7/test_task2.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import pytest
+
+from All_home_works.hw7.task2 import backspace_compare
+
+
+@pytest.mark.parametrize(  # noqa: PT006
```
Comment (Cjkjvfnby) [inline test/hw7/test_task2.py]: What is PT006?

File: test/hw7/test_task2.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "first,second,expected",
+    [
+        ("aB#c", "aD#c", True),
+        ("A##c", "#A#c", True),
+        ("#", "", True),
+        ("abcABCD####", "abc", True),
+    ],
+)
+def test__backspace_compare__with_backspace__expected_true(first, second, expected):
```
Comment (Cjkjvfnby) [inline test/hw7/test_task2.py]: You have `expected_true` in your name. That means that you could simplify this test a bit.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
The solution is good, need to update tests a bit.

File: test/hw7/test_task2.py (hunk 1 above)
Comment (E1-Bart0) [reply test/hw7/test_task2.py]: https://github.com/E1-Bart0/homework-repo/pull/25#discussion_r627397514

File: test/hw7/test_task2.py (hunk 2 above)
Comment (E1-Bart0) [reply test/hw7/test_task2.py]: I have a question to ask you.
If we have True it's recommended to just assert, but if False or None it's better to use assert False and assert None, isn't it?

Comment (Cjkjvfnby) [reply test/hw7/test_task2.py]: > If we have True it's recommended to just assert

Sometimes it is ok to do `assert result is True` just to clarify that you need boolean True explicitly.  But it is not critical. 

After that you have changed the assertion statement, what could you remove?

**Review by Cjkjvfnby — APPROVED** [review]
Bonus task: write this using generator. (O(1) memory)

File: All_home_works/hw7/task2.py [hunk 3]
```diff
@@ -13,18 +13,24 @@
...
+from itertools import zip_longest
 
 
 def backspace_compare(first: str, second: str) -> bool:
-    return get_filtered(first) == get_filtered(second)
+    first_gen = get_filtered(first)
+    second_gen = get_filtered(second)
+    return all(
+        char1 == char2
+        for char1, char2 in zip_longest(first_gen, second_gen, fillvalue=False)
```
Comment (Cjkjvfnby) [inline All_home_works/hw7/task2.py]: `fillvalue=False` it's better to use the value of the same type.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**